### PR TITLE
Group-based board moderation

### PR DESCRIPTION
**Pull request by Oldiesmann**
Signed-off-by: Michael Eshom oldiesmann@oldiesmann.us

Here is the complete group-based board moderation functionality.

Features include:
- Assign/manage moderator groups from the Manage Boards screen
- Moderator groups inherit any additional permissions from the Moderator group on boards they're assigned to moderate
- Moderator groups are listed in the moderators list on display and messageindex
- Moderator groups are listed in the "Boards" report in the admin center
- Users are listed as being able to moderate boards that groups they're in can moderate in the "Staff" report in the admin center
- Groups are shown as having any permissions inherited from "Moderator" in the "Board Permissions" report in the admin center
- Members in groups assigned to moderate a board are treated as if they were a locally-assigned moderator (including being shown as "Moderator" in posts in boards which they can moderate)
- "Track Permissions" page on user's profile shows any additional board-level permissions inherited from Moderator ("Given by: Moderator")

### CODE REVIEW

File: Sources/Reports.php [hunk 1]
```diff
@@ -361,7 +392,7 @@ function BoardPermissionsReport()
 		foreach ($boards as $id => $board)
 			if ($board['profile'] == $row['id_profile'])
 				$board_permissions[$id][$row['id_group']][$row['permission']] = $row['add_deny'];
-
+
```
Comment (emanuele45) [inline Sources/Reports.php]: Spaces! :P

File: Sources/Reports.php [hunk 2]
```diff
@@ -369,7 +400,7 @@ function BoardPermissionsReport()
 			$permissions[$row['permission']] = array(
 				'title' => isset($txt['board_perms_name_' . $row['permission']]) ? $txt['board_perms_name_' . $row['permission']] : $row['permission'],
 			);
-		}
+		}
```
Comment (emanuele45) [inline Sources/Reports.php]: Spaces! :P

File: Sources/Reports.php (hunk 1 above)
Comment (Oldiesmann) [inline Sources/Reports.php]: Whoops! I'll fix that momentarily :)

**Comment by matthew-kerle** [comment]
@joshuaadickerson or I need to get around to throwing this on an installation and giving it a quick test...not against pulling it in though, I personally think it's genius and should definitely have been in the core.

**Comment by XinYenFon** [comment]
I installed this feature and didn't get any error so far my error_log empty too :)

**Pull request merged**